### ISSUE TITLE
`npm view` child fields – escaping keys & square bracket notation

### ISSUE THREAD

**Issue by timoxley** [npm/npm#9207]
### tldr
1. Need way to escape keys like '3.0-next' so they can work with `npm view npm.dist-tags`.
2. `npm view npm.users[0]` square bracket notation doesn't work [as advertised](https://docs.npmjs.com/cli/view).

---

You can use the `npm` cli to query the version of a package like so:

```
> npm info npm dist-tags.next
2.13.5
```

But this doesn't work for dist tags such as npm's own "3.0-next", I assume because it interprets the `.` in `3.0` as part of an object path.

I had hoped there would be some way to escape the dot, so I consulted [the `npm view` documentation](https://docs.npmjs.com/cli/view).

Using square bracket seemed hopeful if it worked like JS square braces.

> You may also use numeric indices in square braces to specifically select an item in an array field. To just get the email address of the first contributor in the list, you can do this:
> 
> `npm view express contributors[0].email`

Bah. This produces no output.

I thought maybe this was because the contributors format changed, I'd try a different array field:

```
npm view express keywords[0]
```

To no avail. No output. 

Note also that the contributors format changing didn't affect the output as this does work: 

```
> npm view express contributors.email
contributors[0].email = 'aaron.heckmann+github@gmail.com'
contributors[1].email = 'ciaranj@gmail.com'
contributors[2].email = 'doug@somethingdoug.com'
contributors[3].email = 'rauchg@gmail.com'
contributors[4].email = 'me@jongleberry.com'
contributors[5].email = 'shtylman+expressjs@gmail.com'
contributors[6].email = 'hanul@hanul.me'
```

**Comment by othiym23** [comment]
Looking at the history of `lib/view.js` doesn't make clear to me when this last worked, if it ever did. Testing with `npm view npm 'contributors[0].email'` fails all the way back to `npm@1.1.62`. It would be nice to have npm follow the same rules that [`json`](http://npm.im/json) does for parsing out the tree, with the nice affordance that things like `contributors.email` provides. Also it would be nice to clean up the code in `lib/view.js` a bit so that it's easier to follow.

The CLI team would like to dedicate some time / a few releases to improving the quality of the CLI's usability, and this is something that might fit in that bucket. That probably isn't going to be for a month or two with what's currently pending, so this might sit around for a bit unless somebody feels like picking this one up and spending some time on it.
